src/main/java/reposense/parser/ArgsParser.java

```diff
@@ -420,8 +424,13 @@ private static void addAnalysisDatesToBuilder(CliArguments.Builder builder, Name
                 ? untilDate
                 : currentDate;
 
-        TimeUtil.verifySinceDateIsValid(sinceDate, currentDate);
-        TimeUtil.verifyDatesRangeIsCorrect(sinceDate, untilDate);
+        if (sinceDate.compareTo(untilDate) > 0) {
+            throw new ParseException(MESSAGE_SINCE_DATE_LATER_THAN_UNTIL_DATE);
+        }
+
+        if (sinceDate.compareTo(currentDate) > 0) {
+            throw new ParseException(MESSAGE_SINCE_DATE_LATER_THAN_TODAY_DATE);
+        }
```

Comment:
Could you add some test coverage for the two new if blocks added?

Reply:
Hi @MarcusTXK, I've added the tests for ArgParser. The PR is ready for review.